### DISCUSSION TITLE
Photo Synchronization(Can't find photos after synchronization)

### DISCUSSION THREAD

**Discussion by juwizeye** [opengisch/QField#2548] [Q&A]
Hi, 
I am testing Qfieldcloud , every thing works fine but I don't know where  pictures which are taken on the field  are stored .In Qfield they  were stored  in project/DCIM  but in Qfieldcloud DCIM folder is not created when I took some folder.
If anyone know where they are stored , I am all ears.

Thanks.

**Comment by suricactus** [comment]
Hey @juwizeye . There are two things to be aware of:
1) make sure your attachement form fields are configured to use relative paths.
2) check the most recent QField versions which fixed a bug about synchronizing attachments.

Duplicate of #2460 #2463

**Reply by enheduanna** [reply]
I have the same issue. I set up the project and also selected photo as attachment but when I take a picture synchronize it and access it on my computer it has not saved the directory in the attribute table but just in photos on the android device, which does not help because the pictures are not labelled. When I do what you @juwizeye suggests - there are two options: relative to default path and relative to project path - which one should be selected.

**Reply by enheduanna** [reply]
P.S. which storage mode should be selected? File path or directory path?

**Reply by juwizeye** [reply]
Hi @enheduanna ,
check the versions of qfieldsync  and Qfield that you are using. It worked for me after that i passed to Qfield 2.0.1 and Qgis3.24.1.
p.s. choose relative path for photos

**Reply by enheduanna** [reply]
Thank you - I am using the 2.0.1 and QGIS 3.24.1 version - just downloaded them again a couple days ago. Maybe install them again? Excuse my ignorance but there are two options "relative to project path" or "relative to default path" and underneath that there is another option for the storage mode "file path" or "directory path". Does it matter which one I choose?

**Reply by juwizeye** [reply]
I think that relative to project path is the best solution. It will add DCIM folder in your projet path. 
![image](https://user-images.githubusercontent.com/96420889/160630574-6d271eb0-f3c7-44c4-910b-283ba02d216f.png)

**Reply by enheduanna** [reply]
Unfortunately I cannot resolve this issue. I have tried each combination - it will not create the DCIM folder unfortunately but only store the photos on the tablet.

**Reply by OBVYamaska** [reply]
If it helps: on some of my android devices, I had to clear the cache of the file manager in the tablet in order to make the DCIM file appear. After that, the pictures were automatically directed there (when the project is correctly configured).

**Comment by enheduanna** [comment]
thank you!

**Comment by juwizeye** [comment]
how do you create your Qfield cloud projet?

**Comment by enheduanna** [comment]
Dear Juwizeye,
I am creating a project in QGIS on my computer. Save the project under a
specific name and then create a cloud project. Then I open the cloud
project in QField and edit the file and push the changes back to the cloud
and then download the cloud project in QGIS on my computer again. It has
all the changes that I make in the attribute table except for the photo.

**Comment by juwizeye** [comment]
how do you create your Qfield cloud project?
![image](https://user-images.githubusercontent.com/96420889/160639433-14dfadd8-5903-4d03-adcd-fcd7f76a5f5d.png)
there are two ways and the folder of your project will change depending on the way that you are using. If you choose the first one it will create a new folder under C:\Users\username\QField\cloud.

**Reply by enheduanna** [reply]
Yes, I use the option that is selected in your image.

**Reply by juwizeye** [reply]
so you should go in C:\Users\yourusername\QField\cloud to see if your projet is not duplicated there.

**Reply by enheduanna** [reply]
<img width="737" alt="Screen Shot 2022-03-29 at 11 08 47 AM" src="https://user-images.githubusercontent.com/31855850/160643820-7bfdffeb-f7f3-4725-883e-0ac7b852ca6c.png">

**Reply by enheduanna** [reply]
is teh USP_Test_Cloud.qps with the little wiggle at the end a duplication?

**Comment by juwizeye** [comment]
you should normally have a folder which is called **.qfieldsync**
![image](https://user-images.githubusercontent.com/96420889/160645183-a01095c9-77e7-4b1d-b88c-e72c8bc9f62a.png)
have you try to see if other modifications that you are doing on Qfield are being done also in your Qgis projet?

**Reply by enheduanna** [reply]
When I change some of the attributes they do show up. Those modifications do show up in QGIS. However what I have noticed is - that if I lets say remove a shp from the project on my computer, try to sync it again to the cloud, they are not properly packaged.

**Reply by enheduanna** [reply]
Meaning I cant reopen the package displaying this recently removed or added shape files.

**Comment by juwizeye** [comment]
what OS are you using ?

**Comment by enheduanna** [comment]
I am using a Mac

**Reply by enheduanna** [reply]
<img width="589" alt="Screen Shot 2022-03-29 at 11 22 48 AM" src="https://user-images.githubusercontent.com/31855850/160646839-0f21d7e7-5f54-48bf-ad23-060ddfefe67d.png">

**Reply by enheduanna** [reply]
but the devices are android lenovo - could there be a compatibility issue

**Comment by juwizeye** [comment]
perhaps that is (macOS) the reason why you don't have .qfieldcloud folder. Me i am using windows. for compatibility with android lenovo, I don't know.

**Reply by enheduanna** [reply]
Dear Juwizeye
I am sorry to bug you but unfortunately I have not received any other response to my other question that I posted on github. I am just wondering if you are able to help. It seems that once you package a project for QField cloud you cannot remove or reload data. Is that correct? At least that is so far my experience - or am I doing something wrong?

I have been setting up a QFieldCloud Project - I first saved the project under a specific name, added all the files and styled them in the way I want them to appear on the cloud and on QField on my devices. However when I remove any of the shp file layers in the QGIS app on my computer, synchronize the QFieldCloud Project and try to add these or other shp to the project on my computer and synchronize the project again, I always get a packaging error. Basically it appears that once I remove a layer I cannot add it again (or not add any files later on, which cannot be true) because it will not package it properly and not display in QField on my android device. The message I get on QField on my tablet is "QFieldCloud had trouble packaging your project" and further below it says for each layer that did not load again after I had removed and reloaded it (or added anew after creating the cloud project) on my computer and synchronized the project to the cloud "Packed layer 'USP_001_Survey Grid' is not valid. Error code invalid_dataprovider, error message"/tmp/tmp294gmyu3/The Umma Survey Project/Field Work Data/ April 2022 Season/USP_001/USP_001 [...] missing." The only workaround is that I delete the current QFieldCloud Project add all the layers I want, package it and then reload the new QFieldCloud Project on my tablet. But it appears as if I cannot remove and reload files from the project itself nor add new files to it and synchronize the project.
Any advice? Would appreciate any help.
Thank you

**Reply by enheduanna** [reply]
sorry was not sure if I posted it in the right box and instead responded to myself

**Reply by Marwe** [reply]
Mac may have the *nix way of interpreting paths, and file-/dirnames starting with a dot like .qfieldcloud there are *hidden*, you need to enable something like *view hidden files* in GUI, or use `ls -a` in a terminal/bash...

**Comment by enheduanna** [comment]
Thank you so much. I am trying it on my pc and see if it resolves the
issue. THanks so much for your help. I really appreciate it. I am leaving
for an archaeological field trip to Iraq in a week and freaking out a bit
because the system does not fully work. Ahhh

**Reply by enheduanna** [reply]
Dear Juwizeye
I am sorry to bug you but unfortunately I have not received any other response to my other question that I posted on github. I am just wondering if you are able to help. It seems that once you package a project for QField cloud you cannot remove or reload data. Is that correct? At least that is so far my experience - or am I doing something wrong? 

I have been setting up a QFieldCloud Project - I first saved the project under a specific name, added all the files and styled them in the way I want them to appear on the cloud and on QField on my devices. However when I remove any of the shp file layers in the QGIS app on my computer, synchronize the QFieldCloud Project and try to add these or other shp to the project on my computer and synchronize the project again, I always get a packaging error. Basically it appears that once I remove a layer I cannot add it again (or not add any files later on, which cannot be true) because it will not package it properly and not display in QField on my android device. The message I get on QField on my tablet is "QFieldCloud had trouble packaging your project" and further below it says for each layer that did not load again after I had removed and reloaded it (or added anew after creating the cloud project) on my computer and synchronized the project to the cloud "Packed layer 'USP_001_Survey Grid' is not valid. Error code invalid_dataprovider, error message"/tmp/tmp294gmyu3/The Umma Survey Project/Field Work Data/ April 2022 Season/USP_001/USP_001 [...] missing." The only workaround is that I delete the current QFieldCloud Project add all the layers I want, package it and then reload the new QFieldCloud Project on my tablet. But it appears as if I cannot remove and reload files from the project itself nor add new files to it and synchronize the project.
Any advice? Would appreciate any help.
Thank you

**Reply by juwizeye** [reply]
Hi, 
 sorry I have been busy. I can delete layers after adding my projet to Qfield cloud but I didn't try to add layers. However, I am using PostgreSQL  database to store my layers .

**Comment by juwizeye** [comment]
try to create a Qfield cloud projet using the second method, but I don't think that it will change something.
Ahh I understand. 
another solution is to download your photos directly from qfield cloud platform but it is tedious task because you will not have them directly in your projet.

**Reply by enheduanna** [reply]
Ah I tried almost everything - qfield cloud just stores it on the local device and it not attached to the file. Its so odd because i did a small mock project to try the system and it worked perfectly without me choosing "relative to the project path"

**Reply by enheduanna** [reply]
sorry I am talking about attachments

**Reply by enheduanna** [reply]
Okay I think it worked. I think the length might have been too short. It was sting and 10 and I think its need to be larger.
